examples/src/main/python/sql/arrow.py

```diff
@@ -32,7 +32,7 @@
 
 
 def dataframe_with_arrow_example(spark):
-    import numpy as np  # type: ignore[import]
+    import numpy as np
```

Comment:
I know this kind of dependency stuff is not very clear in Apache Spark .. but It think we should better keep the ignore for now .. by right NumPy is one of extra dependencies in ML yet .. https://github.com/apache/spark/blob/master/python/setup.py#L262-L274.

Reply:
Thank you for your feedback @HyukjinKwon! 

I am aware that this is an optional dependency, but I don't think it really affects this PR ‒ I will update the description within the next few days to describe different scenarios and possible impact of this change to make that clear.